### PR TITLE
Cleanup of data/simulation analyzer modules in CSCTriggerPrimitives

### PR DESCRIPTION
**Pull request by dildick**
#### PR description:

Following PR https://github.com/cms-sw/cmssw/pull/33085, I add a simple analyzer that overlays all data vs emulator histograms and puts the results into a single postscript file for analysis. 

The analyzers `CSCAnodeLCTAnalyzer`, `CSCAnodeLCTAnalyzer` and `CSCTriggerPrimitivesReader` are being deprecated. The MC-truth analysis is already contained in Validation/MuonHits and Validation/MuonCSCDigis, and included in the standard release validation sequence. The data vs emulator comparisons are now in DQM/L1TEmulator(Client) and DQM/Integration. The remaining useful pieces will be cannibalized into `CSCTriggerPrimitiveAnalyzer`. 
 
I added documentation in README.md files so people know how they can use the test configurations.

#### PR validation:

Tested in CMSSW_12_0_X on a recent run. 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by dildick** [comment]
A few minor bugfixes. I also added configuration to run the offline DQM locally on a data file.

**Comment by silviodonato** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

File: L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc [hunk 1]
```diff
@@ -0,0 +1,204 @@
+#include "L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.h"
```
Comment (silviodonato) [inline L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc:1]: Unit tests to test the functionalities of your Library and/or Plugins should go under .../test.
See https://cms-sw.github.io/cms_coding_rules.html

File: L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
+#ifndef L1Trigger_CSCTriggerPrimitives_CSCTriggerPrimitivesAnalyzer_h
```
Comment (silviodonato) [inline L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.h:1]: Do not split plugins into header and source files.
(see https://cms-sw.github.io/cms_coding_rules.html)

File: L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitivesAnalyzer_cfi.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (silviodonato) [inline L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitivesAnalyzer_cfi.py:1]: Is this code only for testing? If yes, you should move it to `/test`. If not, `The _cfi file should be left to be generated automatically with the fillDescriptions()` (see https://cms-sw.github.io/cms_coding_rules.html and https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Converting_an_Existing_CFI_file)

File: L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+  std::string superDir = monitorDir_.substr(0, pos);
+  monitorDir_.erase(0, pos + delimiter.length());
+  std::string subDir = monitorDir_;
+  std::string path = "DQMData/Run " + std::to_string(runNumber_) + "/" + superDir + "/Run summary/" + subDir + "/";
+
+  std::cout << superDir << " " << subDir << std::endl;
+
+  TFile *theFile = new TFile(rootFileName_.c_str());
+  if (theFile) {
+    std::cout << "Loading DQM ROOT file: " << rootFileName_ << std::endl;
```
Comment (silviodonato) [inline L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc:58]: `cout` is not thread safe, you should use https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger.
You can use `cout` if it is simply a test code

**Review by silviodonato — COMMENTED** [review]
I've added some comments

File: L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc (hunk 4 above)
Comment (dildick) [reply L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc]: Removed or replaced by LogError

File: L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitivesAnalyzer_cfi.py (hunk 3 above)
Comment (dildick) [reply L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitivesAnalyzer_cfi.py]: Moved into the cfg.py file

File: L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.h (hunk 2 above)
Comment (dildick) [reply L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.h]: Turned into a single .cc file.

File: L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc (hunk 1 above)
Comment (dildick) [reply L1Trigger/CSCTriggerPrimitives/plugins/CSCTriggerPrimitivesAnalyzer.cc]: All analyzers moved to test/. Analyzers being deprecated put in test/deprecated/ for now until the remaining useful pieces are integrated

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by qliphy** [comment]
+1

**Pull request merged**